drivers/blktap2.py

```diff
@@ -1410,18 +1410,20 @@ def call_pluginhandler(cls, session, host_ref, sr_uuid, vdi_uuid, action,
 
     def _add_tag(self, vdi_uuid, writable):
         util.SMlog("Adding tag to: %s" % vdi_uuid)
+        params = self.target.vdi.sr.srcmd.params
         attach_mode = "RO"
         if writable:
             attach_mode = "RW"
-        vdi_ref = self._session.xenapi.VDI.get_by_uuid(vdi_uuid)
-        host_ref = self._session.xenapi.host.get_by_uuid(util.get_this_host())
+        vdi_ref = params["vdi_ref"]
+        host_ref = params["host_ref"]
         sm_config = self._session.xenapi.VDI.get_sm_config(vdi_ref)
         attached_as = util.attached_as(sm_config)
         if NO_MULTIPLE_ATTACH and (attached_as == "RW" or \
                 (attached_as == "RO" and attach_mode == "RW")):
             util.SMlog("need to reset VDI %s" % vdi_uuid)
             if not resetvdis.reset_vdi(self._session, vdi_uuid, force=False,
-                    term_output=False, writable=writable):
+                    term_output=False, writable=writable, vdiref=vdi_ref, 
```

Comment:
vdiref, srref, smconf names should be changed: see comment at function definition